core/src/main/java/org/infinispan/transaction/impl/AbstractCacheTransaction.java

```diff
@@ -345,11 +372,6 @@ public long getCreationTime() {
 
    @Override
    public final void addListener(TransactionCompletedListener listener) {
-      notifier.add(listener);
-   }
-
-   @Override
-   public void invoke(TransactionCompletedListener listener) {
-      listener.onCompletion();
+      notifier.thenRunAsync(listener::onCompletion);
```

Comment:
We shouldn't use the async methods without an explicit thread pool parameter, or they'll use the common pool.